public/modules/moduleMetadataService.js

```diff
@@ -1,8 +1,8 @@
 angular.module('newApp')
 .factory('moduleMetadata', function($http) {
     return {
-        get : function() {
-            return $http.get('/api/units');
+        get : function(moduleID) {
+            return $http.get('/api/units' + (moduleID ? '/' + moduleID : ''));
```

Comment:
Yes.  It's an issue of consistency.  Even though javascript is dynamically typed, I've found that you can do yourself a favor by thinking about them anyway.  In this case, the get one call resolves to an object, but the get all call resolves to an array.  Since those methods have different signatures, I generally like to make them different functions.